pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -80,6 +80,58 @@ type ApplicationSelector struct {
 	LabelSelector               *metav1.LabelSelector `json:"labelSelector,omitempty"`
 	metav1.GroupVersionResource `json:",inline"`
 	ResourceRef                 string `json:"resourceRef,omitempty"`
+
+	// BindingPath refers to the path in the application workload's schema
+	// where the binding workload would be referenced.
+	// +optional
+	BindingPath *BindingPath `json:"bindingPath,omitempty"`
+}
+
+var (
+	// DefaultContainersPath has the logical path logical path
+	// to find containers on supported objects
+	// Used as []string{"spec", "template", "spec", "containers"}
+	DefaultPathToContainers = "spec.template.spec.containers"
+
+	// DefaultPathToVolumes is the logical path to find volumes on supported objects
+	// used as []string{"spec", "template", "spec", "volumes"}
+	DefaultPathToVolumes = "spec.template.spec.volumes"
+)
+
+// SetDefaults set default value for binding path
+func (applicationSelector *ApplicationSelector) SetDefaults() {
```

Comment:
@isutton Why do we need to handle that scenario?

Reply:
@baijum to prevent the following error:

```go
// PodSpecPath is nil
applicationSelector.BindingPath = &BindingPath{} 

// Configure application selector defaults
applicationSelector.SetDefaults() 
```

After executing `SetDefaults`, `applicationSelector.BindingPath.PodSpecPath` is still `nil`; this will trigger errors when indiscriminately accessing `applicationSelector.BindingPath.PodSpecPath.Containers`.

Reply:
I have made `volume` optional, but not the `containers`. Should we make that also optional?

Reply:
Hi @baijum I think we need to make that optional. If it is in required and we don't provide an applicationSelector, then the acceptance-test is failing for an `EmptyApplicationSelector` Scenario.

Reply:
I have made `containers` also optional. Let me know if this function still need to be changed.

Reply:
The current code is doing what you explained. The default values are set in `SetDefaults`. There is still a possibility that the application selector could be `nil`. So, when accessing it should be checked for `nil`. Similarly `BindingPath` and `PodSpecPath` also could be `nil`.

Reply:
Done. 8471a6520097095587857d584c3df8585112f11c